src/transforms/pipelines/config.rs

```diff
@@ -41,65 +42,93 @@ impl Clone for PipelineConfig {
     }
 }
 
+#[async_trait::async_trait]
+#[typetag::serde(name = "pipeline")]
+impl TransformConfig for PipelineConfig {
+    async fn build(&self, ctx: &TransformContext) -> crate::Result<Transform> {
+        let condition = if let Some(config) = &self.filter {
+            Some(config.build(&ctx.enrichment_tables)?)
+        } else {
+            None
+        };
+
+        let mut transforms = Vec::new();
```

Comment:
Yeah, there's _a lot_ of this type of thing that should be improved if we seriously go down this path, the main one being allocating a new `TransformOutputsBuf` for each event so that we can do the output/input swap dance as we make our way through the sequence. We may want to introduce a new transform subtype that operates on full batches of events so that we can amortize those kinds of costs (`SyncTransform` kind of has this API but it's awkward and inheritance-y).